tests/integration/protected_images.t

```diff
@@ -0,0 +1,364 @@
+#!/usr/bin/perl -w
+
+use ProductOpener::PerlStandards;
+
+use Test::More;
+use ProductOpener::APITest qw/:all/;
+use ProductOpener::Products qw/:all/;
+use ProductOpener::Test qw/:all/;
+use ProductOpener::TestDefaults qw/:all/;
+
+use File::Basename "dirname";
+
+use Storable qw(dclone);
+
+remove_all_users();
+
+remove_all_products();
+
+wait_application_ready();
+
+# Create an admin
+my $admin_ua = new_client();
+my $resp = create_user($admin_ua, \%admin_user_form);
+ok(!html_displays_error($resp));
+
+# Create a normal user
+my $ua = new_client();
+my %create_user_args = (%default_user_form, (email => 'bob@gmail.com'));
+$resp = create_user($ua, \%create_user_args);
+ok(!html_displays_error($resp));
+
+# Create a moderator
+my $moderator_ua = new_client();
+$resp = create_user($moderator_ua, \%moderator_user_form);
+ok(!html_displays_error($resp));
+
+# Admin gives moderator status
+my %moderator_edit_form = (
+	%moderator_user_form,
+	user_group_moderator => "1",
+	type => "edit",
+);
+$resp = edit_user($admin_ua, \%moderator_edit_form);
+ok(!html_displays_error($resp));
+
+# Create some products
+
+my %product_form = (
+	generic_name => "A generic name",
+	ingredients_text => "apple, milk, eggs, palm oil",
+	categories => "some unknown category",
+	labels => "organic",
+	origin => "france",
+	brands => "Test brand",
+	packaging_text_en =>
+		"1 wooden box to recycle, 6 25cl glass bottles to reuse, 3 steel lids to recycle, 1 plastic film to discard",
+	'nutriment_energy-kj' => 10,
+	nutriment_fat => 5,
+	nutriment_proteins => 2,
+	quantity => "90g",
+	nutrition_facts_image => "image_nutrition",
```

Comment:
Images can't be sent or selected with product_jqm_multilingual.pl or product_multilingual.pl, you will need to use the image upload API: https://openfoodfacts.github.io/openfoodfacts-server/api/ref-v2/#post-/cgi/product_image_upload.pl